docker/templates/Dockerfile.erb

```diff
@@ -175,6 +181,56 @@ WORKDIR /usr/share/logstash
 ENV ELASTIC_CONTAINER true
 ENV PATH=/usr/share/logstash/bin:$PATH
 
+# Add FIPS configuration for observability-sre image flavor
+<% if image_flavor == 'observability-sre' -%>
+
+RUN mkdir -p /usr/share/logstash/config/security
+
+# Copy JVM security configuration files from the unpacked tarball
+RUN cp /usr/share/logstash/x-pack/distributions/internal/observabilitySRE/config/security/java.security /usr/share/logstash/config/security/ && \
+    cp /usr/share/logstash/x-pack/distributions/internal/observabilitySRE/config/security/java.policy /usr/share/logstash/config/security/ && \
+    chown --recursive logstash:root /usr/share/logstash/config/security/
+
+# Convert JKS to BCFKS for truststore and keystore
+RUN /usr/share/logstash/jdk/bin/keytool -importkeystore \
+    -srckeystore /usr/share/logstash/jdk/lib/security/cacerts \
+    -destkeystore /usr/share/logstash/config/security/cacerts.bcfks \
+    -srcstoretype jks \
+    -deststoretype bcfks \
+    -providerpath /usr/share/logstash/logstash-core/lib/jars/bc-fips-2.0.0.jar \
+    -provider org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider \
+    -deststorepass changeit \
```

Comment:
Do we need to have these passwords configurable via an ENV parameter, or similar?

Reply:
I think the desire it to leave it as the default as we are just converting the default. See the convention of using "changeit" here https://docs.oracle.com/javase/10/security/sample-truststores.htm

Reply:
I'm good with this

Reply:
Cool. As soon as i get https://github.com/elastic/logstash/pull/17298 merged i'm going to rebase this branch and get the commit and PR comments all fixed up! thanks.